### PR TITLE
DCKM-579: support for OID4VC import

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: packages/core/src/credentials/oidvc.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  uri,
+  getAuthCode,
+}: {
+  didProvider: IDIDProvider;
+  uri: string;
+  getAuthCode?: (authorizationURL: string) => Promise<string>;
+}) {
+  const [holderKeyDocument] = await didProvider.getDIDKeyPairs();
+
+  const queryString = uri.split('?')[1];
```
Comment (cykoder) [inline packages/core/src/credentials/oidvc.ts:18]: probably cleaner just to do:
```
const params = new URL(uri).searchParams;
```

File: packages/core/src/credentials/oidvc.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const scope = credentialOffer.credentials[0];
+  const format = 'ldp_vc';
+  const credentialTypes = scope.replace('ldp_vc:', '');
+
+  const client = await OpenID4VCIClientV1_0_13.fromURI({
+    uri: uri,
+    clientId: 'dock.wallet',
+    authorizationRequest: {
+      redirectUri: 'dock-wallet://credentials/callback',
+      clientId: 'dock.wallet',
```
Comment (cykoder) [inline packages/core/src/credentials/oidvc.ts:32]: doesnt have to be dock.wallet, could also be `dock-wallet` - maybe this should be a config var (similar for redirect uri above?)

File: packages/core/src/credentials/oidvc.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    },
+  });
+
+  let code;
+
+  if (client.credentialOffer?.preAuthorizedCode) {
+    code = client.credentialOffer?.preAuthorizedCode;
+  } else {
+    code = await getAuthCode(client.authorizationURL); 
+    // TODO: Remove this return
```
Comment (cykoder) [inline packages/core/src/credentials/oidvc.ts:43]: todo now or later?

File: packages/core/src/credential-provider.ts [hunk 4]
```diff
@@ -27,8 +36,39 @@ export function isBBSPlusCredential(credential) {
...
+
+export async function importCredentialFromURI({
+  uri,
+  wallet,
+  didProvider,
+  getAuthCode,
+}: importCredentialFromUriParams & {
+  wallet: IWallet;
+}) {
+  // TODO: unify the impl with the existing import credential flow
```
Comment (cykoder) [inline packages/core/src/credential-provider.ts:54]: todo now or later?

File: packages/core/src/credentials/oidvc.ts (hunk 3 above)
Comment (maycon-mello) [reply packages/core/src/credentials/oidvc.ts:43]: fixed

File: packages/core/src/credential-provider.ts (hunk 4 above)
Comment (maycon-mello) [reply packages/core/src/credential-provider.ts:54]: For later https://dock-team.atlassian.net/browse/DCKM-581

File: packages/core/src/credentials/oidvc.ts (hunk 2 above)
Comment (maycon-mello) [reply packages/core/src/credentials/oidvc.ts:32]: I will handle that in the next PR, I want to unify that with the existing wallet-sdk configs injected by white label apps. Already logged a ticket for it https://dock-team.atlassian.net/browse/DCKM-580

**Review by cykoder — DISMISSED** [review]

**Comment by cykoder** [comment]
lgtm but tests are failing

**Review by cykoder — APPROVED** [review]

**Pull request merged**